src/biotite/structure/charges.py

```diff
@@ -138,6 +137,78 @@
 # electronegativity values)
 EN_POS_HYDROGEN = 20.02
 
+
+def _determine_aromatic_nitrogen_hybridisation(
+    elements, types, bond_types
+):
+    """
+    Determine whether a nitrogen atom involved in an aromatic system
+    possesses exclusively single bonds or one double bond as well.
+
+    The need to determine this arises from the fact that the assignment
+    of parameters for electronegativity computation is based on the
+    hybridisation state, i. e. no special parameters for the involvement
+    in aromatic systems are available. This poses no problem in the case
+    of the elements oxygen and sulfur as they are exclusively involved
+    in aromatic systems having single bonds on either side, enabling the
+    the assignment of aromaticity to a distinct hybridisation state.
+    Nitrogen, however, can be involved in aromatic systems having both
+    exclusively single bonds as well as one double bond, making the
+    assignment of aromaticity to a distinct hybridisation state
+    impossible. Examples for nitrogen atoms involved in an aromatic
+    system having exclusively single bonds as well as one double bond
+    within one and the same molecule are the aromatic compounds
+    'imidazole' and 'pyrazole'.
+
+    Parameters
+    ----------
+    elements: ndarray, dtype=str
+        The array comprising the elements of the inserted `atom_array`.
+    types: ndarray, shape=(n,m), dtype=int
+        The array containing the bond types of each bond of every atom.
+    bond_types: ndarray, dtype=int
+        The array containing information about the highest bond type of
+        the respective atom (except for BondType.AROMATIC).
+
+    Returns
+    -------
+    bond_types: ndarray, dtype=int
+        The inserted array in which the BondType.AROMATIC is replaced by
+        the a BondType revealing the bond order in case of nitrogen
+        atoms.
+    """
+
+    nitrogen_indices = np.where(elements == "N")[0]
+    for i in nitrogen_indices:
+         if BondType.AROMATIC in types[i]:
```

Comment:
Please combine with boolean mask from above for higher performance and rename variable accordingly:
```python
aromatic_nitrogen_indices = np.where((elements == "N") & (types == BondType.AROMATIC))[0]
```

Reply:
Done.